src/spyglass/decoding/v1/core.py

```diff
@@ -239,10 +235,31 @@ def _upsample(
             np.unique(np.concatenate((position_df.index, new_time))),
             name="time",
         )
+
+        # Find NaN intervals
+        nan_intervals = {}
+        for column in position_df.columns:
+            is_nan = position_df[column].isna().values.astype(int)
```

Comment:
Why the conversion to int?

Reply:
Otherwise the `np.diff` calls in the next lines will be true when the bool changes in either direction without differentiating.  Want to be able to quickly identify where nan's start and stop